nautobot/core/views/__init__.py

```diff
@@ -284,29 +296,54 @@ def collect(self):
         yield gauge
 
 
-def nautobot_metrics_view(request):
-    """Exports /metrics.
+class PrometheusVersioning(URLPathVersioning):
```

Comment:
The URL doesn't include the version, but when accessing via curl/telegraf, it is exposed in the headers as Content-Type. Telegraf in particular is very picky that the version matches and will return a 406 if the version doesn't match.

Telegraf uses `application/vnd.google.protobuf;proto=io.prometheus.client.MetricFamily;encoding=delimited;q=0.7,text/plain;version=0.0.4;q=0.3` for it's accept header when using the prometheus input plugin. This can be replicated with curl though.

In the `NautobotMetricsView` if you comment out the `versioning_class` line (line 318), and do a curl:
```
curl -H "Accept: application/vnd.google.protobuf;proto=io.prometheus.client.MetricFamily;encoding=delimited;q=0.7,text/plain;version=0.0.4;q=0.3" http://dev01.home.gotz.co:8080/metrics/ -v
```

Detail of the curl message:
```
curl -H "Accept: application/vnd.google.protobuf;proto=io.prometheus.client.MetricFamily;encoding=delimited;q=0.7,text/plain;version=0.0.4;q=0.3" http://localhost:8080/metrics/ -v
*   Trying 192.168.74.40:8080...
* TCP_NODELAY set
* Connected to localhost (192.168.74.40) port 8080 (#0)
> GET /metrics/ HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/7.68.0
> Accept: application/vnd.google.protobuf;proto=io.prometheus.client.MetricFamily;encoding=delimited;q=0.7,text/plain;version=0.0.4;q=0.3
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 406 Not Acceptable
< Date: Thu, 08 Feb 2024 20:16:12 GMT
< Server: WSGIServer/0.2 CPython/3.8.18
< Content-Type: text/plain; charset=utf-8
< Allow: GET, HEAD, OPTIONS
< X-Content-Type-Options: nosniff
< Referrer-Policy: same-origin
< X-Frame-Options: DENY
< Content-Length: 133
< Vary: Cookie, origin
< djdt-store-id: 2f95eb63d9cb46b180ed11e9d1170f37
< Server-Timing: TimerPanel_utime;dur=88.41800000000433;desc="User CPU time", TimerPanel_stime;dur=21.850999999998066;desc="System CPU time", TimerPanel_total;dur=110.26900000000239;desc="Total CPU time", TimerPanel_total_time;dur=124.58432512357831;desc="Elapsed time", SQLPanel_sql_time;dur=6.156029179692268;desc="SQL 1 queries", CachePanel_total_time;dur=0;desc="Cache 0 Calls"
<
* Connection #0 to host localhost left intact
{'detail': ErrorDetail(string='Invalid version "0.0.4" in "Accept" header. Supported versions are: 2.0, 2.1', code='not_acceptable')}%
```

Reply:
noticed the BaseVersioning vs URLPathVersioning. I'll test that part out

Reply:
Looks like it needs to be `AcceptHeaderVersioning`